>From f7ad06a3e88d9b7e5a7f37932d726dc6cfcb25fc Mon Sep 17 00:00:00 2001
From: Chow Loong Jin <hyperair@debian.org>
Date: Sat, 8 Jun 2013 22:09:39 +0800
Subject: [PATCH] Fix dh-exec-subst-env handling of empty vars

When an environment variable is set to an empty variable or 0, dh-exec ignores
it and restores the ${foo} placeholder, but this shouldn't be the case.

This commit checks for the existence of the variable in %ENV instead of just
using || for falling back.
---
 lib/dh-exec-subst-env | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/dh-exec-subst-env b/lib/dh-exec-subst-env
index 312490e..940c39e 100755
--- a/lib/dh-exec-subst-env
+++ b/lib/dh-exec-subst-env
@@ -4,4 +4,4 @@
 ## corresponding environment variable, or the original ${...} if the
 ## env. var does not exist.
 $_ = "" if /^#/;
-s/(\$\{([^\}]+)\})/$ENV{$2} || $1/eg;
+s/(\$\{([^\}]+)\})/exists $ENV{$2} ? $ENV{$2} : $1/eg;
-- 
1.8.1.2

